policies/branch-protection-sdks.yml

```diff
@@ -0,0 +1,59 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+name: branch_protection_sdks
+description: Organization branch protection policy for Microsoft Graph SDKs
+resource: repository
+where:
+  - |
+    repository.name.startsWith("msgraph-sdk-", StringComparison.InvariantCultureIgnoreCase)
+    || repository.name.startsWith("msgraph-beta-sdk-", StringComparison.InvariantCultureIgnoreCase)
+    || repository.name.equals("msgraph-cli", StringComparison.InvariantCultureIgnoreCase)
+    || repository.name.equals("msgraph-cli-core", StringComparison.InvariantCultureIgnoreCase)
+    || repository.name.equals("msgraph-beta-cli", StringComparison.InvariantCultureIgnoreCase)
+    || repository.name.equals("msgraph-beta-typescript-typings", StringComparison.InvariantCultureIgnoreCase)
+    || repository.name.equals("msgraph-typescript-typings", StringComparison.InvariantCultureIgnoreCase)
+    || repository.name.equals("microsoft-graph-toolkit", StringComparison.InvariantCultureIgnoreCase)
+configuration:
+  branchProtectionRules:
+  - branchNamePattern: main
+    requiredApprovingReviewsCount:
+      min: 1
+    # Must have a CODEOWNER approve for the PR to be merged.
+    requireCodeOwnersReview: true
+    # Require status checks to pass before merging. TODO: this value should be true, we should work to support this.
+    # Used with the requiredStatusChecks setting to specify which checks must pass for the PR to be merged.
+    requiresStrictStatusChecks: false
+    # TODO: all commits should be signed. We need to get everyone signing their commits.
```

Comment:
let me know if you want me to drop my setup script somewhere in a doc

Reply:
Can you provide me with a link to your script so I can add it to my planning doc?

Reply:
```bash
#this line is only for windows, we'll need to document the equivalent for linux and mac
winget install GnuPG.Gpg4win
#generate a certificate, the email must match for the line below, upload the public key to github sign keys, get the certificate thumbprint. For all those steps, see the blog post below
git config user.email <microsoft-email>
git config user.signingkey <certificateThumbprint>
git config user.name <FirstNameLastName>
git config commit.gpgsign true
git config tag.gpgsign true
# this line is only needed for windows
git config --global gpg.program "C:\Program Files (x86)\GnuPG\bin\gpg.exe"
# for linux/mac if the $PATH is setup to include gpg, it should not be needed
```
> Note: add --global to configure for all repos

https://jamesmckay.net/2016/02/signing-git-commits-with-gpg-on-windows/